_posts/2023-12-11-augmented-reality-in-travel-applications.md

```diff
@@ -0,0 +1,130 @@
+  // the logic
+});
+```
+
+To mark locations visually on the camera, we make use of an `a-box`, which we add to the DOM. With the help of attributes, we can determine its position. We also determine what is rendered. For the first test, I simply opted for blue bars. To inform the scene about the location, we dispatch the event `gps-entity-place-loaded`.
+
+```js
+const icon = document.createElement("a-box");
+icon.setAttribute(
+  "gps-entity-place",
+  `latitude: ${latitude};longitude: ${longitude}`
+);
+icon.setAttribute("name", place.name);
+icon.setAttribute("width", "100");
+icon.setAttribute("height", "35");
+icon.setAttribute("material", "color: #0e55cd");
+
+icon.addEventListener("loaded", () =>
+  window.dispatchEvent(new CustomEvent("gps-entity-place-loaded"))
+);
+```
+
+We have already created a dynamic: The locations are rendered in the camera, during the movement. The only thing missing is interaction.
+
+We want the user to be shown the name of the point of interest and its distance from the current location (which we have calculated beforehand) when they click on the object:
+
+```js
+icon.addEventListener("click", (e) => {
+  e.stopPropagation();
+  e.preventDefault();
+
+  const location = e.target;
+  const locationName = location.getAttribute("name");
+
+  alert(locationName + "\nDistance: " + distance + " meters");
+});
+```
+
+The majority is once again: HTML. Therefore, we must now add the `a-box` elements to the scene:
+
+```js
+const scene = document.querySelector("a-scene");
+// ...
+scene.appendChild(icon);
+```
+
+And this was the first working version:
+
+<figure>
+    <video src="https://media-cdn.holidaycheck.com/video/upload/v1702298405/videos/blog/elfarxu9nrcjkjpemvvg.mp4"></video>
+    <figcaption>The first version of the interactive camera with blue blocks marking the position of the location.</figcaption>
+</figure>
+
+Admittedly, this doesn't look very appealing. But it shows very well how the different locations are displayed in terms of size. While the Fuggerei is closer to my current location, it appears larger than the town hall, which is shown smaller because it is further away. That's super cool!
+
+<figure>
+    <video src="https://media-cdn.holidaycheck.com/video/upload/v1702298408/videos/blog/yhjwymyfd2mtoh8sazn6.mp4"></video>
+    <figcaption>The second version of the interactive camera, which renders the name and a picture of the location.</figcaption>
+</figure>
+
+## Conclusion
+
+This experiment was more of an opportunity to broaden my horizons and test something new, to play with it. Doing this in a product context made it even more exciting.
+
+Fast-forward 2 years, through our explorations at HolidayCheck, we've seen firsthand how the synergy of AI and AR can revolutionize travel experiences. In this context, I would also like to thank Konark Modi, who, with his expertise in AI, is always on the lookout on how technology can further help our Urlaubers.
```

Comment:
You used `Urlaubers` here, but `vacationers` previously.
I'd suggest sticking to one :v:

Reply:
"Urlaubers" is the way to go 😎